src/esrs/store.rs

```diff
@@ -134,8 +134,8 @@ pub struct StoreEvent<Event> {
 }
 
 impl<Event> StoreEvent<Event> {
-    pub const fn sequence_number(&self) -> SequenceNumber {
-        self.sequence_number
+    pub const fn sequence_number(&self) -> &SequenceNumber {
+        &self.sequence_number
```

Comment:
I know it's copy, but here I'm getting the number OF self, so I like the type to contain a reference to self.
You can dereference it to copy if you need a free-standing value, detached from self.